CBForest/slice.hh

```diff
@@ -69,6 +69,8 @@ namespace forestdb {
 
         explicit operator std::string() const;
         std::string hexString() const;
+        
+        bool hasPrefix(const char* str) const;
```

Comment:
It would be more general purpose if the parameter type were a slice. It's easy to convert a `char*` to a slice (there's a constructor.)